src/core/utils.ts

```diff
@@ -9,8 +9,11 @@ export { clone, deepAssign, log, riot, GbTracker };
 
 export const WINDOW = () => window;
 
+const ARRAY_TO_DOT_NOTATION_REGEX = /\[(\d+)\]/g;
```

Comment:
thanks for taking my regex and actually making it what we want (for some reason I was thinking we needed to support doubles too :stuck_out_tongue_closed_eyes:)